services/galley/src/Galley/Intra/Effects.hs

```diff
@@ -147,25 +152,47 @@ interpretDefederationNotifications ::
   Sem (DefederationNotifications ': r) a ->
   Sem r a
 interpretDefederationNotifications = interpret $ \case
-  SendDefederationNotifications domain -> do
-    maxPage <- inputs $ fromRange . currentFanoutLimit . _options -- This is based on the limits in removeIfLargeFanout
-    page <- embedClient $ paginate selectAllMembers (paramsP LocalQuorum () maxPage)
-    void $ sendNotificationPage page
-    where
-      pushEvents results = do
-        let (bots, mems) = localBotsAndUsers results
-            recipients = Intra.recipient <$> mems
-            event = Intra.FederationEvent $ Federation.Event Federation.FederationDelete domain
-        for_ (Intra.newPush ListComplete Nothing event recipients) $ \p -> do
-          -- Futurework: Transient or not?
-          -- RouteAny is used as it will wake up mobile clients
-          -- and notify them of the changes to federation state.
-          push1 $ p & Intra.pushRoute .~ Intra.RouteAny
-        deliverAsync (bots `zip` repeat (G.pushEventJson event))
-      sendNotificationPage page = do
-        let res = result page
-            mems = mapMaybe toMember res
-        pushEvents mems
-        when (hasMore page) $ do
-          page' <- embedClient $ nextPage page
-          sendNotificationPage page'
+  SendDefederationNotifications domain ->
+    getPage
+      >>= void . sendNotificationPage (Federation.FederationDelete domain)
+  SendOnConnectionRemovedNotifications domainA domainB ->
```

Comment:
I think we should only send the client notification if we actually do federate with both `domainA` and `domainB`. I am not sure if this is actually verified somewhere. It might be and I am currently not seeing it :)

Reply:
I've updated this to check that the receiving Galley instance is federated with both domains. If it is, it will process the request. If it isn't, the request is ignored. I've also updated the tests, as they weren't setting up the federations, and now they require that.